cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -38,17 +38,38 @@ func NewAzureResourceManager(azCli azcli.AzCli) *AzureResourceManager {
 	}
 }
 
+var hasTargetResourceData = func(operation *armresources.DeploymentOperation) bool {
+	return operation.Properties.TargetResource != nil
+}
+
+var isDeploymentType = func(operation *armresources.DeploymentOperation) bool {
+	return *operation.Properties.TargetResource.ResourceType == string(AzureResourceTypeDeployment)
```

Comment:
Should we have `nil` guards here to ensure that `operation.Properties.TargetResource.ResourceType` is non nil?

Reply:
Similar questions about `hasResourceType` and `isCreateOperation`